controllers/kuadrant_controller.go

```diff
@@ -181,44 +181,40 @@ func (r *KuadrantReconciler) unregisterExternalAuthorizer(ctx context.Context) e
 
 func (r *KuadrantReconciler) unregisterExternalAuthorizerIstio(ctx context.Context) error {
 	logger, _ := logr.FromContext(ctx)
-	iop := &iopv1alpha1.IstioOperator{}
+	var configsToUpdate []istio.ConfigWrapper
 
+	iop := &iopv1alpha1.IstioOperator{}
 	iopKey := client.ObjectKey{Name: controlPlaneProviderName(), Namespace: controlPlaneProviderNamespace()}
 	if err := r.Client().Get(ctx, iopKey, iop); err != nil {
 		logger.V(1).Info("failed to get istiooperator object", "key", iopKey, "err", err)
-		return err
+		if apimeta.IsNoMatchError(err) {
+			// return err only if there's no istiooperator CRD
+			return err
+		}
+		// otherwise, we assume that the control plane is istio but no installed by its operator
+	} else {
+		configsToUpdate = append(configsToUpdate, istio.NewOperatorWrapper(iop))
 	}
 
-	meshConfig, err := meshConfigFromStruct(iop.Spec.MeshConfig)
-	if err != nil {
+	istioConfigMap := &corev1.ConfigMap{}
+	if err := r.Client().Get(ctx, client.ObjectKey{Name: "istio", Namespace: controlPlaneProviderNamespace()}, istioConfigMap); err != nil {
+		logger.V(1).Info("failed to get istio configMap", "key", iopKey, "err", err)
 		return err
 	}
-	extensionProviders := extensionProvidersFromMeshConfig(meshConfig)
-
-	if !hasKuadrantAuthorizer(extensionProviders) {
-		return nil
-	}
+	configsToUpdate = append(configsToUpdate, istio.NewConfigMapWrapper(istioConfigMap))
 
-	for idx, extensionProvider := range extensionProviders {
-		name := extensionProvider.Name
-		if name == extAuthorizerName {
-			// deletes the element in the array
-			extensionProviders = append(extensionProviders[:idx], extensionProviders[idx+1:]...)
-			meshConfig.ExtensionProviders = extensionProviders
-			meshConfigStruct, err := meshConfigToStruct(meshConfig)
-			if err != nil {
+	for _, config := range configsToUpdate {
+		if needsUpdate, err := config.UpdateConfig(
+			istio.RemoveKuadrantAuthorizerFromConfig,
+		); err != nil {
+			return err
+		} else if needsUpdate {
+			logger.Info("remove external authorizer from meshconfig")
+			if err := r.Client().Update(ctx, config.GetConfigObject()); err != nil {
```

Comment:
I'm guessing your intention was of keeping like before, to not change too much. So just pointing out we have this alternative from the BaseReconciler:

```suggestion
			if err := r.UpdateResource(ctx, config.GetConfigObject()); err != nil {
```